components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/dao/impl/TopicDAOImpl.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.websubhub.publisher.dao.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.database.utils.jdbc.NamedJdbcTemplate;
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.identity.event.websubhub.publisher.constant.TopicSQLConstants;
+import org.wso2.identity.event.websubhub.publisher.dao.TopicDAO;
+import org.wso2.identity.event.websubhub.publisher.exception.WebSubAdapterException;
+import org.wso2.identity.event.websubhub.publisher.model.Topic;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.UUID;
+
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.ErrorMessages.ERROR_TOPIC_PERSISTENCE;
+import static org.wso2.identity.event.websubhub.publisher.constant.WebSubHubAdapterConstants.ErrorMessages.ERROR_TOPIC_RETRIEVAL;
+import static org.wso2.identity.event.websubhub.publisher.util.WebSubHubAdapterUtil.handleServerException;
+
+/**
+ * Implementation of the TopicDAO interface.
+ * This class handles all the database operations related to WebSubHub topic management.
+ */
+public class TopicDAOImpl implements TopicDAO {
+
+    private static final Log log = LogFactory.getLog(TopicDAOImpl.class);
+
+    @Override
+    public String addTopic(Topic topic) throws WebSubAdapterException {
+
+        String id = UUID.randomUUID().toString();
```

Comment:
This is not something the DAO layer should do. DAO should just persist and retrieve